light-zk.js/src/transaction/prover.ts

```diff
@@ -128,7 +128,7 @@ export class Prover {
     const res = await snarkjs.groth16.verify(
       this.vKey,
       this.publicInputs,
-      this.proof,
+      this.proof || proof,
```

Comment:
I think it should be the other way around?

If `proof` is not `null`, we should use it, otherwise fall back to `this.proof`. `this.proof` can be never `null`, because we have a non-null assertion (`public proof!: proofData;`in line 37). So this would be probably more correct:

```suggestion
      proof || this.proof,
```

But also... I don't really like the fact of having that optional `proof?` parameter here, it's very confusing and I couldn't find any place in our code which would actually provide it. I think defining the proof always in the constructor should be the way. @ananas-block any reason for introducing that parameter here?

If there is no reason, let's just remove it?

Reply:
@xonoxitron thanks good catch!

@vadorovsky agreed had the same thought if proof is provided it should be used over this.proof.

There can be a case that we don't generate the proof with the prover class thus this.proof can be undefined.
For example in the relayer we want to verify the proof of the transaction we are relaying prior to executing the transaction.
In this case we want to insert a proof that was not generated by the prover class instance.
To me it feels more intuitive to pass the proof in the verify function than in the constructor.

@xonoxitron what is your intuition whether the proof should be passed in the constructor or the verify function?
Once the order is changed happy to merge!

Reply:
it is used here: https://github.com/xonoxitron/light-protocol/blob/main/light-zk.js/src/transaction/transaction.ts#L381

Reply:
> @xonoxitron thanks good catch!
> 
> @vadorovsky agreed had the same thought if proof is provided it should be used over this.proof.
> 
> There can be a case that we don't generate the proof with the prover class thus this.proof can be undefined. For example in the relayer we want to verify the proof of the transaction we are relaying prior to executing the transaction. In this case we want to insert a proof that was not generated by the prover class instance. To me it feels more intuitive to pass the proof in the verify function than in the constructor.
> 
> @xonoxitron what is your intuition whether the proof should be passed in the constructor or the verify function? Once the order is changed happy to merge!

thanks @ananas-block.

IMHO I agree on the fact of removing the optional "?" from the "proof" input variable.